hypothesis-python/src/hypothesis/internal/compat.py

```diff
@@ -43,7 +42,9 @@ def int_to_byte(i):
 def qualname(f):
     try:
         return f.__qualname__
-    except AttributeError:
+    except AttributeError:  # pragma: no cover
+        # It's unclear whether this fallback is actually needed by anything,
+        # but it probably doesn't hurt either.
```

Comment:
Hmm, I should come back and clean this up in a future PR.

Nontrivial refactoring effort though, as it was a common problem under py2, but now we have qualname logic smeared across compat, reflection, the ghostwriter, etc.  In some places `.__qualname__` is always safe and we can inline it.  See `git grep -E '\bqualname\(.+\)'`.